kinematics_base/include/moveit/kinematics_base/kinematics_base.h

```diff
@@ -103,6 +148,22 @@ class KinematicsBase
                              const kinematics::KinematicsQueryOptions &options = kinematics::KinematicsQueryOptions()) const = 0;
 
   /**
+   * @brief Given a desired pose of the end-effector, compute the set joint angles solutions that are able to reach it.  This
+   * is a default implementation that returns only one solution and so its result is equivalent to calling 'getPositionIK(...)'
+   * with a zero initialized seed.
+   * @param ik_pose the desired pose of the tip frame
+   * @param solutions a vector of vectors where each entry is a valid solution
+   * @param result a struct that reports the results of the query
+   * @param an option struct which contains the type of redundancy discretization used.  This default implementation only
+   * supports the KinmaticSearches::ONE discretization; requesting any other will result in failure.
+   * @return True if a valid set of solutions was found, false otherwise
+   */
+  virtual bool getMultipleIK(const geometry_msgs::Pose &ik_pose,
```

Comment:
It may be rather difficult to support multiple tips through this method because it'd be unclear to deternime which solutions (more than 1 per pose) belong to each tip pose.